### PR TITLE
test: fix metrics again

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 

as title

### What is changed and how it works?

To meet the demanding conditions of the test scenario
only events like `WriteRowsEventV2` are now handled in the code by using failpoint

### Check List 

Tests 

 - Integration test

Code changes

Side effects

 - Increased code complexity

Related changes

### CODE REVIEW

**Comment by lance6716** [comment]
what's the query that caused the failure? I think ignoring query will fail the data synchronization so test can't pass check_diff

**Comment by Ehco1996** [comment]
> what's the query that caused the failure? I think ignoring query will fail the data synchronization so test can't pass check_diff

in here https://github.com/pingcap/dm/pull/2030#issuecomment-904211995

It does not affect the result of check_diff because only one dml statement will be written in the final stage of the test, and this dml statement can be processed normally

**Comment by lance6716** [comment]
> > what's the query that caused the failure? I think ignoring query will fail the data synchronization so test can't pass check_diff
> 
> in here [#2030 (comment)](https://github.com/pingcap/dm/pull/2030#issuecomment-904211995)
> 
> It does not affect the result of check_diff because only one dml statement will be written in the final stage of the test, and this dml statement can be processed normally

What's the content of query? DDL of course will change data thus failed the check_diff

**Comment by Ehco1996** [comment]
@lance6716  add more fix in 
https://github.com/pingcap/dm/pull/2033/commits/12b6671f4039a2c9b54f75231b94966957ec63ff

this should cover most of the scenarios PTAL

File: syncer/syncer.go [hunk 1]
```diff
@@ -1804,6 +1804,13 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 			return terror.ErrSyncerGetEvent.Generate(err)
 		}
 
+		failpoint.Inject("OnlySomeTypeEventCanPass", func(val failpoint.Value) {
```
Comment (lance6716) [inline syncer/syncer.go:1807]: we should only skip "HeartbeatEvent". IMO we should revert this change of failpoint

File: tests/metrics/run.sh [hunk 2]
```diff
@@ -99,22 +99,25 @@ function run() {
...
 	# At 6s, a dml job is added to job chan and the ticker is reset
 	# At 11s the ticker write the log of the second no job
 	# Check that the interval between the two ticker logs is > 5s
+
+	rm -rf $WORK_DIR/worker1/log/dm-worker.log # clean up the old log
 	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
 	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
 
 	echo "sleep 6s"
 	sleep 6
```
Comment (lance6716) [inline tests/metrics/run.sh:119]: maybe this sleep is no longer need, because check_log_contain_with_retry will find a proper sleep time

**Comment by Ehco1996** [comment]
another unsteable test
```

[2021-08-24T07:43:32.784Z] go: downloading github.com/pingcap/check v0.0.0-20200212061837-5e12011dc712

[2021-08-24T07:43:32.784Z] go: downloading github.com/kami-zh/go-capturer v0.0.0-20171211120116-e492ea43421d

[2021-08-24T07:45:24.394Z] ?   	github.com/pingcap/dm/chaos/cases	[no test files]

[2021-08-24T07:45:24.394Z] ok  	github.com/pingcap/dm/checker	0.394s	coverage: 76.6% of statements

[2021-08-24T07:45:24.394Z] ?   	github.com/pingcap/dm/debug-tools/binlog-event-blackhole	[no test files]

[2021-08-24T07:45:24.394Z] ok  	github.com/pingcap/dm/dm/common	0.041s	coverage: 64.1% of statements

[2021-08-24T07:45:24.394Z] ok  	github.com/pingcap/dm/dm/config	0.535s	coverage: 71.5% of statements

[2021-08-24T07:45:24.394Z] ?   	github.com/pingcap/dm/dm/ctl	[no test files]

[2021-08-24T07:45:24.394Z] ?   	github.com/pingcap/dm/dm/ctl/common	[no test files]

[2021-08-24T07:45:24.394Z] ok  	github.com/pingcap/dm/dm/ctl/master	0.137s	coverage: 2.7% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/dm/master	98.395s	coverage: 62.2% of statements

[2021-08-24T07:47:15.815Z] ?   	github.com/pingcap/dm/dm/master/metrics	[no test files]

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/dm/master/scheduler	22.327s	coverage: 81.2% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/dm/master/shardddl	99.587s	coverage: 76.7% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/dm/master/workerrpc	0.263s	coverage: 84.6% of statements

[2021-08-24T07:47:15.815Z] ?   	github.com/pingcap/dm/dm/pb	[no test files]

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/dm/portal	0.204s	coverage: 68.8% of statements

[2021-08-24T07:47:15.815Z] ?   	github.com/pingcap/dm/dm/portal/statik	[no test files]

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/dm/unit	0.214s	coverage: 100.0% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/dm/worker	94.499s	coverage: 71.8% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/dumpling	10.345s	coverage: 72.5% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/loader	0.207s	coverage: 29.7% of statements

[2021-08-24T07:47:15.815Z] ?   	github.com/pingcap/dm/monitoring/dashboards	[no test files]

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/backoff	0.041s	coverage: 100.0% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/conn	0.221s	coverage: 69.9% of statements

[2021-08-24T07:47:15.815Z] ?   	github.com/pingcap/dm/pkg/context	[no test files]

[2021-08-24T07:47:15.815Z] ?   	github.com/pingcap/dm/pkg/cputil	[no test files]

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/dumpling	0.311s	coverage: 74.7% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/election	7.364s	coverage: 86.9% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/encrypt	0.055s	coverage: 91.2% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/etcdutil	13.467s	coverage: 81.6% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/func-rollback	0.039s	coverage: 100.0% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/gtid	0.054s	coverage: 85.0% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/ha	26.954s	coverage: 70.2% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/helper	0.062s	coverage: 100.0% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/log	0.054s	coverage: 78.0% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/metricsproxy	0.047s	coverage: 100.0% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/parser	0.270s	coverage: 88.7% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/retry	7.290s	coverage: 81.2% of statements

[2021-08-24T07:47:15.815Z] ok  	github.com/pingcap/dm/pkg/schema	27.203s	coverage: 85.0% of statements

[2021-08-24T07:47:15.816Z] ok  	github.com/pingcap/dm/pkg/shardddl/optimism	33.531s	coverage: 73.4% of statements

[2021-08-24T07:47:15.816Z] ok  	github.com/pingcap/dm/pkg/shardddl/pessimism	2.537s	coverage: 81.1% of statements

[2021-08-24T07:47:15.816Z] 

[2021-08-24T07:47:15.816Z] ----------------------------------------------------------------------

[2021-08-24T07:47:15.816Z] FAIL: reader_test.go:259: testReaderSuite.TestParseFilerelayLogUpdatedOrNewCreated

[2021-08-24T07:47:15.816Z] 

[2021-08-24T07:47:15.816Z] reader_test.go:296:

[2021-08-24T07:47:15.816Z]     c.Assert(err, IsNil)

[2021-08-24T07:47:15.816Z] ... value *errors.withStack = decode relay meta toml file failed: open /tmp/check-1171447680058747669/4/b60868af-5a6f-11e9-9ea3-0242ac160006.000001/relay.meta: no such file or directory ("decode relay meta toml file failed: open /tmp/check-1171447680058747669/4/b60868af-5a6f-11e9-9ea3-0242ac160006.000001/relay.meta: no such file or directory")
```

**Review by lance6716 — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM
